src/librustc_infer/infer/mod.rs

```diff
@@ -1681,13 +1683,14 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
     /// having to resort to storing full `GenericArg`s in `stalled_on`.
     #[inline(always)]
     pub fn ty_or_const_infer_var_changed(&self, infer_var: TyOrConstInferVar<'tcx>) -> bool {
+        let mut inner = self.inner.borrow_mut();
```

Comment:
Why these changes? In general, increasing the scope of `borrow_mut` seems risky

Reply:
Seemed like a simplification to me, the function itself doesn't recurse, it only checks the type/const variable so there is no way `borrow_mut` may be called again.